sys/kern/kern_proc.c

```diff
@@ -2803,44 +2835,67 @@ kern_proc_vmmap_out(struct proc *p, struct sbuf *sb, ssize_t maxlen, int flags)
 			kve->kve_type = KVME_TYPE_NONE;
 			kve->kve_ref_count = 0;
 			kve->kve_shadow_count = 0;
+
+			/* Anonymous memory is presumed able to have tags */
+			kve->kve_protection |= KVME_PROT_LOADTAGS
+					       | KVME_PROT_STORETAGS;
+
+			kve->kve_max_protection |= KVME_PROT_LOADTAGS
+						   | KVME_PROT_STORETAGS;
 		}
 
-		strlcpy(kve->kve_path, fullpath, sizeof(kve->kve_path));
+		if ((kve->kve_protection & KVME_PROT_READ) == 0)
+			kve->kve_protection &= ~KVME_PROT_LOADTAGS;
+		if ((kve->kve_protection & KVME_PROT_WRITE) == 0)
+			kve->kve_protection &= ~KVME_PROT_STORETAGS;
+		if ((kve->kve_max_protection & KVME_PROT_READ) == 0)
+			kve->kve_max_protection &= ~KVME_PROT_LOADTAGS;
+		if ((kve->kve_max_protection & KVME_PROT_WRITE) == 0)
+			kve->kve_max_protection &= ~KVME_PROT_STORETAGS;
```

Comment:
Isn't this lying?

Reply:
I suppose so, even if those are the implications enforced by our model.  I guess we can leave it to userland to understand the implications instead.